service/src/main/java/uk/nhs/adaptors/gp2gp/common/storageconnectors/StorageConnectorFactory.java

```diff
@@ -0,0 +1,30 @@
+package uk.nhs.adaptors.gp2gp.common.storageconnectors;
+
+import static uk.nhs.adaptors.gp2gp.common.storageconnectors.StorageConnectorOptions.AZURE_BLOB;
+import static uk.nhs.adaptors.gp2gp.common.storageconnectors.StorageConnectorOptions.S3;
+
+public class StorageConnectorFactory {
+
+    private static StorageConnector storageConnector;
+
+    public static StorageConnector getConfiguredConnector() {
+        String storageMode = System.getenv("STORAGE_SOLUTION");
```

Comment:
This environmental variable key could also be stored in config. What do you think?

Reply:
Good use of the factory pattern here! The next step is to use it as a spring factory bean. That will also let us inject the configuration values from application.yml as @mmackiewicz1 suggested.